airbyte-config/init/src/main/java/io/airbyte/config/init/RemoteDefinitionsProvider.java

```diff
@@ -55,16 +55,20 @@ public RemoteDefinitionsProvider(@Value("${airbyte.platform.remote-connector-cat
   }
 
   @PostConstruct
-  public void initialize() throws InterruptedException, IOException {
-    final CombinedConnectorCatalog catalog = getRemoteDefinitionCatalog(this.remoteDefinitionCatalogUrl, this.timeout);
-    this.sourceDefinitions = catalog.getSources().stream().collect(Collectors.toMap(
-        StandardSourceDefinition::getSourceDefinitionId,
-        source -> source.withProtocolVersion(
-            AirbyteProtocolVersion.getWithDefault(source.getSpec() != null ? source.getSpec().getProtocolVersion() : null).serialize())));
-    this.destinationDefinitions = catalog.getDestinations().stream().collect(Collectors.toMap(
-        StandardDestinationDefinition::getDestinationDefinitionId,
-        destination -> destination.withProtocolVersion(
-            AirbyteProtocolVersion.getWithDefault(destination.getSpec() != null ? destination.getSpec().getProtocolVersion() : null).serialize())));
+  public void loadDefinitions() {
```

Comment:
Add a @CachePut("remoteDefinitions") annotation here instead of `@PostConstruct`

all the getters in this provider should call `loadDefinitions()`